DEV_GUIDE.md

```diff
@@ -0,0 +1,163 @@
+#Developer guide
+
+##Contributing new content
+
+In addition to including features / fixes, you can contribute new content to Scala Exercises! The platform is designed in a way that creating new exercises is pretty straight-forward. Currently there are several libraries available in production (i.e.: STD lib, Cats, Shapeless...), and more are to be included soon. If you're interested in creating a new exercises library, you'll find all the needed steps in this document.
+
+###Overview
+
+The `scala-exercises` organization holds different repositories for both the core project ("scala-exercises"), and each of the main exercises libraries (i.e.: "exercises-stdlib", "exercises-cats"...). The core project is currently built on one single module, but it'll be potentially splitted up in several ones in a not-so-distant future. The content you can provide to the platform doesn't necessarily be in the previously mentioned organization, but for them to appear inside the main `scala-exercises` in production, they need to be published in a public artifact repository like Sonatype or Maven-central. All the steps, from creating the content library to the publication process, are described in the following sections.
```

Comment:
`The core project is currently built on one single module` no need to mention this and not completely accurate at this time